crates/gen/src/element_type.rs

```diff
@@ -0,0 +1,165 @@
+use crate::blob::Blob;
```

Comment:
It wasn't quite clear to me where the best place for the ElementType implementation was, so I settled on here, but please tell me if there's a better place!